BSM2 Flowsheet Documentation Update

MarcusHolly opened the pull request:
## Fixes/Resolves:

Addresses feedback on BSM2 flowsheet documentation by adding sections for DOFs, flowsheet specifications, additional variables, additional constraints, and the flowsheet's objective function.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

MarcusHolly committed:
Make refinements to BSM2 documentation

MarcusHolly committed:
Minor touch-ups

MarcusHolly committed:
Update extended BSM2 documentation

MarcusHolly committed:
Minor touch-up

agarciadiego commented on docs/technical_reference/flowsheets/BSM2.rst:
```diff
@@ -52,6 +61,131 @@ Documentation for each of the property models can be found here:
+    * dewatering unit hydraulic retention time and specific energy consumption
+    * thickener hydraulic retention time and diameter
+
+Flowsheet Specifications
```
can we add any information from where our specifications came from?

MarcusHolly replied:
I've added the references used for the feed water concentrations.

agarciadiego reviewed (approved):
LGTM just a comment for adding additional context, but can be merged if there is no real answer

MarcusHolly committed:
Add references for influent wastewater characteristics

adam-a-a commented on docs/technical_reference/flowsheets/extended_BSM2.rst:
```diff
@@ -41,6 +41,15 @@ The flowsheet relies on the following key assumptions:
 
+The objective function is to minimize the levelized cost of water, which can be represented by the following equation
+where :math:`Q` represents volumetric flow, :math:`f_{crf}` represents capital recovery factor
+:math:`C_{ca,tot}` represents total capital cost, :math:`C_{op,tot}` represents total operating cost, and
```
```suggestion
:math:`C_{cap,tot}` represents total capital cost, :math:`C_{op,tot}` represents total operating cost, and
```

adam-a-a replied:
My suggestion is to apply this to both rst files.

adam-a-a commented on docs/technical_reference/flowsheets/extended_BSM2.rst:
```diff
@@ -41,6 +41,15 @@ The flowsheet relies on the following key assumptions:
+
+    .. math::
+
+        LCOW_{Q} = \frac{f_{crf}   C_{ca,tot} + C_{op,tot}}{f_{util} Q}
```
```suggestion
        LCOW_{Q} = \frac{f_{crf}   C_{cap,tot} + C_{op,tot}}{f_{util} Q}
```

adam-a-a replied:
My suggestion is to apply this to both rst files.

adam-a-a commented on docs/technical_reference/flowsheets/extended_BSM2.rst:
```diff
@@ -56,6 +65,143 @@ Documentation for each of the property models can be found here:
 
+Degrees of Freedom
+------------------
+The following variables must be specified by the user to run the BSM2 flowsheet:
```
```suggestion
The following variables must are initially specified for simulating the BSM2 flowsheet (i.e., degrees of freedom = 0):
```

adam-a-a replied:
My suggestion is to apply this to both rst files.

MarcusHolly committed:
Merge branch 'main' into bsm2_doc_update

adam-a-a commented on docs/technical_reference/flowsheets/extended_BSM2.rst:
```diff
@@ -56,6 +65,143 @@ Documentation for each of the property models can be found here:
+   "Reactor 5 mass transfer", ":math:`injection_{R5, S_{O2}} = KLa_{R5} * V_{R5} * (S_{O, eq} - S_{O, out})`"
+   "Reactor 6 mass transfer", ":math:`injection_{R6, S_{O2}} = KLa_{R6} * V_{R6} * (S_{O, eq} - S_{O, out})`"
+   "Reactor 7 mass transfer", ":math:`injection_{R7, S_{O2}} = KLa_{R7} * V_{R7} * (S_{O, eq} - S_{O, out})`"
+
```
Still unsure whether we should include this Future refinements section. On the one hand, it let's the user know that the flowsheet isn't perfect and points out particular areas of improvement. On the other hand, we'll have to remember to update this section whenever relevant changes are applied, and I can imagine flowsheets where no future refinement is planned. The alternative would just be to track future refinements in an issue.

Let's see if @TimBartholomew or others have thoughts on this.

MarcusHolly committed:
Address Adam's comments

adam-a-a commented on docs/technical_reference/flowsheets/BSM2.rst:
```diff
@@ -52,6 +61,131 @@ Documentation for each of the property models can be found here:
+
+Additional Constraints
+----------------------
+The subscript "out" represents the effluent from the secondary clarifier
```
```suggestion
The following constraints represent concentration limits on treated water (secondary effluent) discharged into the environment. The subscript "out" represents the effluent from the secondary clarifier.
```

adam-a-a reviewed (changes requested):
Just a couple of suggested changes. I also think we should clarify the number of reactors in the activated sludge process for each flowsheet (i.e., 5 and 7 for conventional and extended, respectively). 

With that, we may want to update the extended BSM2 diagram to reflect the additional reactors. Perhaps @luohezhiming could assist with that.

MarcusHolly committed:
Update BSM2 figures

MarcusHolly committed:
Clarify the number of each reactor type in implementation

MarcusHolly commented:
> Just a couple of suggested changes. I also think we should clarify the number of reactors in the activated sludge process for each flowsheet (i.e., 5 and 7 for conventional and extended, respectively).
> 
> With that, we may want to update the extended BSM2 diagram to reflect the additional reactors. Perhaps @luohezhiming could assist with that.

@luohezhiming I've updated the BSM2 figures slideshow in Box and in this PR. Basically I've added an image for extended BSM2 and added the stirrers to the CSTR blocks, but feel free to edit them again later as you see fit.

adam-a-a commented:
@agarciadiego thinks that Future refinements section should be an optional section so that users can find our planned improvements (more transparency than issues) and reach out if interested in stated plans or new feature requests.

@luohezhiming also agrees that we should keep as an optional section.

adam-a-a commented on docs/technical_reference/flowsheets/BSM2.rst:
```diff
@@ -37,6 +37,15 @@ The flowsheet relies on the following key assumptions:
+    .. math::
+
+        LCOW_{Q} = \frac{f_{crf}   C_{cap,tot} + C_{op,tot}}{f_{util} Q}
+
```
Add a documentation section for unit costing

MarcusHolly replied:
See lines 54-64 in the BSM2 flowsheet documentation. I've moved the WaterTAP costing package link to this sectoin as well as the objective function.

adam-a-a commented on docs/technical_reference/flowsheets/BSM2.rst:
```diff
@@ -37,6 +37,15 @@ The flowsheet relies on the following key assumptions:
+        LCOW_{Q} = \frac{f_{crf}   C_{cap,tot} + C_{op,tot}}{f_{util} Q}
+
 Documentation for each of the unit models can be found here:
     * `Thickener <https://watertap.readthedocs.io/en/latest/technical_reference/unit_models/thickener.html>`_
```
List config option details for unit. If default config options used, can explain with generic sentence

MarcusHolly replied:
See L39 for the generic sentence (all units in BSM2 use their default configurations)

adam-a-a commented on docs/technical_reference/flowsheets/BSM2.rst:
```diff
@@ -52,6 +61,132 @@ Documentation for each of the property models can be found here:
     * `ASM1 <https://watertap.readthedocs.io/en/latest/technical_reference/property_models/ASM1.html>`_
```
I think I replaced CSTR_with_injection with AerationTank, which inherits from CSTR with injection but sets certain config options.

adam-a-a replied:
If so, we should list Aeration tank instead of CSTR w/injection

MarcusHolly replied:
Corrected - thanks for catching this!

MarcusHolly committed:
Merge branch 'main' into bsm2_doc_update

MarcusHolly committed:
Address Tim's feedback

MarcusHolly committed:
Correct typo

lbianchi-lbl committed:
Merge branch 'main' into bsm2_doc_update
